syntax = "proto3";

import "google/protobuf/any.proto";

message MethodPB {
  string name = 1;
  repeated ParameterPB args = 2;
}


message ParameterPB {
  string name = 1;
  ParameterTypePB type = 2;
  string annotation = 3;
  google.protobuf.Any default_value = 4;
  google.protobuf.Any value = 5;
}


enum ParameterTypePB {
  DEFAULT = 0;
  POSITIONAL = 1;
  KEYWORD = 2;
  VAR_POSITIONAL = 3;
  VAR_KEYWORD = 4;
}